contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -67,8 +65,20 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 		logger.Infof("Received message from %q channel", channel.String())
 		// publish to Natss
 		ch := getSubject(channel)
+		if s.natssConn == nil {
```

Comment:
When is the lock required to be held to access or mutate `s.natssConn`?

Reply:
The reason I did not put lock here is because the code which is setting this value is in the same goroutine. I tested this code with `go tests --race` it is all clean.

Reply:
I don't think I follow your explanation. The only place I see `s.natssConn` set is `connectWithRetry`, is that running in this goroutine?

Based on the comment above, we only need to hold the lock while writing `s.natssConn`, is that why we don't grab the lock here?